python/ray/serve/backend_state_manager.py

```diff
@@ -977,12 +995,17 @@ def update(self) -> bool:
             slow_start_replicas = []
             for replica in replicas.pop(
                     states=[ReplicaState.STARTING_OR_UPDATING]):
-                ready, slow_start = replica.check_started()
+                (ready, slow_start,
+                 exception_in_constructor) = replica.check_started()
                 if ready:
                     # This replica should be now be added to handle's replica
                     # set.
                     replicas.add(ReplicaState.RUNNING, replica)
                     transitioned_backend_tags.add(backend_tag)
+                elif exception_in_constructor:
+                    # TODO(jiaodong): Probably better to handle it at replica level,
```

Comment:
Maybe do:
```
                    replica.set_should_stop(0)
                    replicas.add(ReplicaState.SHOULD_STOP, replica)
```

You may need to add some sort of 'if already dead logic' to `BackendReplica::stop`

Reply:
I think deleting just the replica is a good idea because it is definitely possible for a constructor to fail at first, but succeed at a later time.

Reply:
Yep I tried this for an earlier PR as well but it didn't play nicely without leveraging asyncio.Future to set exceptions. Let me give it another try, it's definitely preferable to only take action on replica level rather than shutting down entire deployment.

Reply:
So there needs more changes in addition to marking its state as SHOULD_STOP since we still thinks target_replica should be 1, thus still causes it to loop forever. Let me send a followup commit to further change target_replica to 0, and let update() shut it down through controller's event loop.